nextstep-mvc/src/main/java/nextstep/mvc/DispatcherServlet.java

```diff
@@ -1,57 +1,77 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
+import java.util.List;
 
 @WebServlet(name = "dispatcher", urlPatterns = "/", loadOnStartup = 1)
 public class DispatcherServlet extends HttpServlet {
-    private static final long serialVersionUID = 1L;
     private static final Logger logger = LoggerFactory.getLogger(DispatcherServlet.class);
-    private static final String DEFAULT_REDIRECT_PREFIX = "redirect:";
+    private static final long serialVersionUID = 1L;
 
-    private HandlerMapping rm;
+    private List<HandlerResolver> handlerResolvers;
+    private List<ViewResolver> viewResolvers;
 
-    public DispatcherServlet(HandlerMapping rm) {
-        this.rm = rm;
+    public DispatcherServlet(List<HandlerResolver> handlerAdapters, List<ViewResolver> viewResolvers) {
+        this.handlerResolvers = handlerAdapters;
+        this.viewResolvers = viewResolvers;
     }
 
     @Override
     public void init() throws ServletException {
-        rm.initialize();
+        handlerResolvers.forEach(HandlerResolver::initialize);
     }
 
     @Override
-    protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String requestUri = req.getRequestURI();
-        logger.debug("Method : {}, Request URI : {}", req.getMethod(), requestUri);
-
-        Controller controller = rm.getHandler(requestUri);
+    protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        HandlerExecution handler = mappingHandler(req, resp);
+        ModelAndView mav = handle(req, resp, handler);
+        View view = resolveView(mav);
         try {
-            String viewName = controller.execute(req, resp);
-            move(viewName, req, resp);
-        } catch (Throwable e) {
-            logger.error("Exception : {}", e);
-            throw new ServletException(e.getMessage());
+            view.render(mav.getModel(), req, resp);
+        } catch (Exception e) {
+            logger.debug(e.getMessage(), e.getCause());
+            throw new ViewRenderException();
         }
     }
 
-    private void move(String viewName, HttpServletRequest req, HttpServletResponse resp)
-            throws ServletException, IOException {
-        if (viewName.startsWith(DEFAULT_REDIRECT_PREFIX)) {
-            resp.sendRedirect(viewName.substring(DEFAULT_REDIRECT_PREFIX.length()));
-            return;
+    private HandlerExecution mappingHandler(HttpServletRequest req, HttpServletResponse resp) {
+        return handlerResolvers.stream().filter(resolver -> resolver.support(req, resp))
+                .findAny()
+                .orElseThrow(IllegalAccessError::new)
+                .getHandler(req)
+                ;
+    }
+
+    private ModelAndView handle(HttpServletRequest req, HttpServletResponse resp, HandlerExecution handlerExecution) {
```

Comment:
```suggestion
    private ModelAndView handle(HttpServletRequest req, HttpServletResponse resp, HandlerExecution handlerExecution) {
        try {
            return handlerExecution.handle(req, resp);
        } catch (Exception e) {
            logger.debug("message : {}, cause : {}", e.getMessage(), e.getCause());
            throw new RuntimeException(e);
        }
    }
```
이렇게 하면 로컬 변수를 선언하지 않아도 될 것 같네요

Reply:
감사합니다 피드백 적용했습니다!